warehouse/query-core/src/main/java/datawave/query/jexl/visitors/FixUnfieldedTermsVisitor.java

```diff
@@ -206,21 +206,23 @@ protected void visit(JexlNode node, JexlNode newNode, Object data) {
                         // Only add that node if it actually has children
                         if (0 < newChild.jjtGetNumChildren()) {
                             newNode.jjtAddChild(newChild, newIndex);
+                            newChild.jjtSetParent(newNode);
                             newIndex++;
                         }
                     } else {
                         // Otherwise, we want to add the child regardless
                         newNode.jjtAddChild(newChild, newIndex);
+                        newChild.jjtSetParent(newNode);
                         newIndex++;
                     }
                 }
             } else {
                 
                 newNode.jjtAddChild(node.jjtGetChild(i), newIndex);
+                node.jjtSetParent(newNode);
```

Comment:
Is this supposed to be `node.jjtGetChild(i).jjtSetParent(newNode);`?

Reply:
I suspect your are correct. Interestingly enough, after dropping a System.exit(0) here the build still passes...so yay for test coverage